### PR TITLE
Local development

### PR DESCRIPTION
**Pull request by r-czajkowski**
Add docker-compose file to run the graph node locally and update the readme file- describe how to deploy the subgraph locally and test against the local chain.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -114,3 +114,47 @@ https://api.studio.thegraph.com/query/24143/threshold/0.0.2
...
+the development network config in
+https://github.com/threshold-network/solidity-contracts/blob/main/hardhat.config.ts#L42-L44[hardhat.config.ts]
+is correct for your local chain.
+
+Run `yarn deploy --network development --reset` to deploy contracts to your
+local chain.
+
+=== Update the subgraph manifest
+
+Set the correct `addrres` and `startBlock` for contracts in `subgraph.yaml`
```
Comment (manumonti) [inline README.adoc]: [typo] addrres -> address

**Review by manumonti — COMMENTED** [review]
Just a little typo in README

File: README.adoc (hunk 1 above)
Comment (r-czajkowski) [reply README.adoc]: 4d28a9e

File: README.adoc [hunk 2]
```diff
@@ -139,3 +139,47 @@ https://api.studio.thegraph.com/query/24143/threshold/0.0.3
...
+=== Run local chain eg. Ganache
+
+Note that host set to `0.0.0.0` is necessary for Ganache to be accessible from
+within Docker and from other machines. By default, Ganache only binds to
+127.0.0.1, which can only be accessed from the host machine that Ganache runs
+on. If you use a ganache-cli run the node with `ganache-cli -h 0.0.0.0`. If you
+use a Ganche in GUI version go to `settings -> server ` and set `HOSTNAME` to
+`0.0.0.0 - All Interfaces`.
+
+=== Deploy the Threshold Network contracts to local chain
```
Comment (manumonti) [inline README.adoc:157]: Following these steps are not enough to have running the local Graph Node. In addition to these, we should write here that, in order to deploy the local Graph Node, it's necessary:
- To have `keep-core` contracts deployed (and how to do this).
OR
- To modify deployment scripts of `solidity-contract` repo to deploy the stubs.

IMHO these steps are necessary to be written here, especially for those of us who are not familiar with keep-core contracts deployment.

Comment (r-czajkowski) [reply README.adoc:157]: 22da4ff

File: README.adoc [hunk 3]
```diff
@@ -139,3 +139,72 @@ https://api.studio.thegraph.com/query/24143/threshold/0.0.3
...
+https://github.com/threshold-network/solidity-contracts/blob/main/hardhat.config.ts#L42-L44[hardhat.config.ts]
+is correct for your local chain.
+
+NOTE: The deployment scripts assume the `keep-core` contract are already
+deployed to your local chain.
+
+==== Deploy the Keep Core contracts to local chain
+
+Clone https://github.com/keep-network/keep-core repo and make sure the
+`development` network config in
```
Comment (manumonti) [inline README.adoc]: `development` is correct? or do you mean `local`?

File: README.adoc [hunk 4]
```diff
@@ -139,3 +139,72 @@ https://api.studio.thegraph.com/query/24143/threshold/0.0.3
...
+https://github.com/keep-network/keep-core/blob/main/solidity-v1/truffle-config.js#L8-L12[truffle-config.js]
+is correct for your local chain.
+
+Go to root dir and run `./scripts/install.sh --network local --contracts-only `.
```
Comment (manumonti) [inline README.adoc]: typo: there is a whitespace character that prevent to render the line correctly: 
[...] --contracts-only_`.

**Review by manumonti — COMMENTED** [review]
Only some typos.

File: README.adoc [hunk 5]
```diff
@@ -139,3 +139,72 @@ https://api.studio.thegraph.com/query/24143/threshold/0.0.3
...
+=== Run local chain eg. Ganache
+
+Note that host set to `0.0.0.0` is necessary for Ganache to be accessible from
+within Docker and from other machines. By default, Ganache only binds to
+127.0.0.1, which can only be accessed from the host machine that Ganache runs
+on. If you use a ganache-cli run the node with `ganache-cli -h 0.0.0.0`. If you
+use a Ganche in GUI version go to `settings -> server ` and set `HOSTNAME` to
+`0.0.0.0 - All Interfaces`.
+
+=== Deploy the Keep Core contracts to local chain
```
Comment (manumonti) [inline README.adoc]: This line is repeated in line 166.

**Comment by manumonti** [comment]
Please, don't forget to add in README that chainID and networkID in ganache must be the same.

File: README.adoc (hunk 5 above)
Comment (r-czajkowski) [reply README.adoc]: 4d2092e

File: README.adoc (hunk 4 above)
Comment (r-czajkowski) [reply README.adoc]: 4d2092e

File: README.adoc (hunk 3 above)
Comment (r-czajkowski) [reply README.adoc]: 4d2092e

**Review by manumonti — APPROVED** [review]
Nice work!

**Pull request merged**